hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -2174,21 +2174,14 @@ public static HoodieMetadataColumnStats mergeColumnStatsRecords(HoodieMetadataCo
       return newColumnStats;
     }
 
-    Comparable minValue =
-        (Comparable) Stream.of(
-                (Comparable) unwrapAvroValueWrapper(prevColumnStats.getMinValue()),
-                (Comparable) unwrapAvroValueWrapper(newColumnStats.getMinValue()))
-            .filter(Objects::nonNull)
-            .min(Comparator.naturalOrder())
-            .orElse(null);
-
-    Comparable maxValue =
-        (Comparable) Stream.of(
-                (Comparable) unwrapAvroValueWrapper(prevColumnStats.getMaxValue()),
-                (Comparable) unwrapAvroValueWrapper(newColumnStats.getMaxValue()))
-            .filter(Objects::nonNull)
-            .max(Comparator.naturalOrder())
-            .orElse(null);
+    Comparable minValue = castAndCompare(
+        unwrapAvroValueWrapper(newColumnStats.getMinValue()),
+        unwrapAvroValueWrapper(prevColumnStats.getMinValue()),
+        true);
```

Comment:
Is this for supporting schema evolution?

Reply:
We need to think more about type promotion since the column and partition stats may need to be regenerated in this case for correctness.  For example, take a list of values of `[1,2,3,4,5,6,7,8,9,10]`, when promoting the type from integer to String, the min-max range is changed from `[1,10]` to `["1", "9"]`, which cannot be done through simple type cast.  I think it's better to avoid the type cast here and handle the stats merge somewhere else to avoid possible corruption of stats.